### ISSUE TITLE
🐛 Bug Report: Getting Started seems broken at the moment

### ISSUE THREAD

**Issue by ericfrederich** [backstage/backstage#27403]
### 📜 Description

Following [getting started](https://backstage.io/docs/getting-started/) guide I am running into errors on Ubuntu 24.04 under WSL.

### 👍 Expected behavior

I shouldn't get an error running `npx @backstage/create-app@latest`

### 👎 Actual Behavior with Screenshots

Note: I re-ran the `npx @backstage/create-app@latest` command to get this screenshot because I had already closed the terminal.
![image](https://github.com/user-attachments/assets/190d599c-5765-4b7f-bccd-0a47ba4c1c59)

### 👟 Reproduction steps

Here's my complete history from a fresh install of Ubuntu-24.04
```
    1  sudo nano -w /etc/wsl.conf 
    2  echo "${LOGNAME} ALL=(ALL) NOPASSWD:ALL" | sudo tee /etc/sudoers.d/${LOGNAME}
    3  sudo apt update
    4  sudo apt upgrade -y
    5  sudo apt install -y build-essential docker.io
    6  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash
    7  echo "here in a new terminal after installing nvm"
    8  nvm install 22
    9  npm install --global yarn
   10  mkdir src && cd src
   11  npx @backstage/create-app@latest
   12  cat /tmp/xfs-8cf2eabe/build.log | sed "s#${LOGNAME}#ubuntu-user#g" > /mnt/c/Users/${LOGNAME}/Desktop/build.log
   13  history | sed "s#${LOGNAME}#ubuntu-user#g" > /mnt/c/Users/${LOGNAME}/Desktop/history
```

### 📃 Provide the context for the Bug.

I get an error with the following log.
[build.log](https://github.com/user-attachments/files/17577664/build.log)

Here's what seems to be the actual compile-time error.
```
../src/isolate/class_handle.h: In lambda function:
../src/isolate/class_handle.h:324:39: error: there are no arguments to ‘RuntimeTypeError’ that depend on a template parameter, so a declaration of ‘RuntimeTypeError’ must be available [-fpermissive]
  324 |                                 throw RuntimeTypeError(detail::CalleeName(info)+ " must be called with `new`");
      |                                       ^~~~~~~~~~~~~~~~
```

### 🖥️ Your Environment

This was done using a fresh install of Ubuntu 24.04 on WSL2.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by ericfrederich** [comment]
So the getting started guide says: `Node.js [Active LTS Release](https://nodejs.org/en/about/previous-releases) installed using one of these methods ...

After clicking that link I saw that 22 is the latest LTS.  For the hell of it I tried the same install with 20 instead of 22 and it seemed to work.

Is backstage not compatible with Node.js 22?

**Comment by darylgraham** [comment]
Backstage support has been for Node 18 until recently when it was bumped up to 20. I believe 20 is the active LTS and 22 is latest LTS, which is where the confusion lies.

**Comment by camilaibs** [comment]
Hi @Rugvip, could you help clarify? I saw you bumped node version last week, so I think you'd be the best person to reply: https://github.com/backstage/backstage/pull/27286.

**Comment by camilaibs** [comment]
Related: https://github.com/backstage/backstage/issues/27396

**Comment by Rugvip** [comment]
Full details are here: https://backstage.io/docs/overview/versioning-policy#nodejs-releases

TL;DR: The switch to the new LTS versions is not immediate but requires a release for it to be rolled out. It does not affect the current release. The actual supported versions for each release are what is specified in `engines` in the root `package.json`.

I think the proper fix here might be to for example include the last couple of lines from the `yarn install` failure, because right now we're hiding the underlying error when it fails.

**Comment by awanlin** [comment]
Agree with @Rugvip comments above around improvements to the output. That would make it a lot easier for new adopters to know what to do and for people who are trying to help out with issues.

As a potential smaller improvement I've created #27415 to update the link on the Getting Started page to point to the Node version policy as it makes this more clear: https://backstage.io/docs/overview/versioning-policy#nodejs-releases

@ericfrederich, could you look at that PR and let me know if that would have helped? 

@darylgraham, same, might help clarify your understanding as we do support the two current Active LTS versions but there is clearly a gap during the change when a new version becomes Active LTS. I added a sentence to try and make this more clear so your eyes on that would be great 👍

**Comment by TimKnight-DWP** [comment]
This all sounds great as long as the docs/errors communicate what the state of play is, it'd prevent new users such as myself in #27626 and OP here being confused 👍

**Comment by vinzscam** [comment]
I just want to add that the next Backstage mainline release which should fix the issue is scheduled for November 19th.
